### PR TITLE
Added is_Empty Set

### LINKED ISSUE #13956: Add is_EmptySet for Interval
#### Reference to related PR's

See #11501 

#### What needs to be done?

Complete the PR #11501 . Addressing the comments left by @siefkenj would suffice I guess.
The contributor shouldn't forget to have `Fixes #11501 and Fixes #13956` in PR description.

### PR DESCRIPTION
**Pull request by arighnac**
Fixes #13956 .
Fixes #11501 .

This PR implements is_EmptySet() in `Interval` class in sets.py.

### CODE REVIEW

**Comment by gxyd** [comment]
You should follow this comment, as was mentioned in the issue description https://github.com/sympy/sympy/pull/11501#issuecomment-268026696 .

**Comment by ArighnaIITG** [comment]
@gxyd, @ylemkimon 
It has been implemented along @siefkenj 's comment.
Please review.

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -940,6 +938,11 @@ def right_open(self):
         """
         return self._args[3]
 
+    @property
+    def is_EmptySet(self):
+        if not (self.end == self.start and (self.left_open or self.right_open)):
+            return (self.end - self.start).is_negative
```
Comment (siefkenj) [inline sympy/sets/sets.py]: The logic here still seems wrong. What does `(a,a)` return? By my read it returns `None` when it should return `True`.

**Comment by jksuom** [comment]
This attribute has already been defined for EmptySet and also for intervals that are empty.
```
>>> EmptySet.is_EmptySet
True
>>> Interval.open(a, a)
EmptySet()
>>> Interval.open(a, a).is_EmptySet
True
>>> Interval.Lopen(a, a).is_EmptySet
True
>>> Interval.Ropen(a, a).is_EmptySet
True
>>> Interval(1, 0).is_EmptySet
True
```

**Comment by ArighnaIITG** [comment]
@siefkenj , I think this has already been addressed in the test case added.
@jksuom, Do I have to implement anymore?

**Comment by jksuom** [comment]
I see no reason for an implementation of `is_EmptySet` in the `Interval` class as an empty interval *is* `EmptySet`.

**Comment by ArighnaIITG** [comment]
So, was the previous implementation correct? @jksuom

**Comment by jksuom** [comment]
> was the previous implementation correct?

If you mean [this implementation](https://github.com/sympy/sympy/blob/master/sympy/sets/sets.py#L1772), then I think it is correct.

**Comment by ArighnaIITG** [comment]
Yes, but then this PR and related issues make no sense?

**Comment by jksuom** [comment]
In SymPy, expressions of type `x.is_Class` usually mean that `x` an instance of class `Class`. So one would expect `J.is_EmptySet` to be `True` if and only if `J` really is `EmptySet`, and therefore `J.is_EmptySet` should not be `True` for an instance `J` of class `Interval`. It could be `False`, at least in those cases where the interval can be show to be non-empty. This is what could be implemented here.

**Comment by gxyd** [comment]
Since currently:
```python
In [1]: a = Interval(x, y)

In [2]: a.is_FiniteSet
Out[2]: False
```

That would also imply that `a.is_EmptySet` should be `False` (considered mathematically, since a non-finite set can't be empty set).
Also when considered from sympy's point (as you explain), `a.is_EmptySet` should be `False` (since, if `a` can never be an instance of `FiniteSet`, neither can it be of `EmptySet`).

TBH, I don't get real picture of the usability of `is_Class` (even when we tried to introduce in our finitely presented groups PR, then Sartaj also raised the point of simply avoid using it and instead use `isinstance`).

Is there any distinction between `isinstance(a, EmptySet)` (or `is S.EmptySet`) and `.is_EmptySet`? Is that something related to `isinstance` also including `a` being an instance of (assumed) super-class of `EmptySet`? While `.is_EmptySet` always (atleast with that aim) checking when `a` is exactly an instance of `EmptySet` and not of some superclass of `EmptySet`? (similar to `type(a) is EmptySet`)

**Comment by jksuom** [comment]
I think that a more useful attribute/property would be `is_empty`. That could be a `Set` class attribute with default value `None`. For `EmptySet` it would redefined as `True`. Other classes could return `False` when that can be verified (and otherwise `None`).

**Comment by ArighnaIITG** [comment]
@jksuom,
So will I define a property is_empty in the `Interval` class? And what classes will I check in it?

**Comment by jksuom** [comment]
> what classes will I check in it?

You could probably start with a few (maybe one or two) classes where it could be seen (in certain cases) that an instance is not empty, `Interval` and `Union`, for example.

**Comment by ArighnaIITG** [comment]
There has been a issue. I had previously messed up my master branch, and thus had to clone the entire repository. So there has been a merge conflict, which I believe can only be solved by creating another branch upon the new master. So I am closing this PR.

**Pull request closed without merging**